core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1482,9 +1490,23 @@ private void handleSqlBasicCallForTimestampMulti(SqlWriter writer, SqlCall call)
   private void unparseExtractFunction(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
     switch (call.operand(0).toString()) {
     case "EPOCH" :
-      final SqlWriter.Frame epochFrame = writer.startFunCall("UNIX_SECONDS");
-      call.operand(1).unparse(writer, leftPrec, rightPrec);
-      writer.endFunCall(epochFrame);
+      SqlNode firstOperand = call.operand(1);
+      if (firstOperand instanceof SqlBasicCall
+          && ((SqlBasicCall) firstOperand).getOperator().kind == SqlKind.MINUS) {
+        SqlNode leftOperand = ((SqlBasicCall) firstOperand).getOperands()[0];
```

Comment:
Shall we add check for timestamp operand as DATE is not supported for now?

Reply:
When the operand will be for suppose : extract(epoch from birth_date - current_date())
In this Case this will become extract('epoch' from Date_diff(birth_date - current_date()))

In first place it will not even enter into epich case and will go to default case due to wrong translation from mig.
so adding that check is of no use for now.